src/cswinrt/code_writers.h

```diff
@@ -3549,6 +3550,452 @@ private % AsInternal(InterfaceTag<%> _) => % ?? Make_%();
+                                    {
+                                        w.write("    %Marshaller.ConvertToManaged(value.%)", td.TypeName(), fieldName);
+                                    }
+                                    else
+                                    {
+                                        w.write("    value.%", fieldName);
+                                    }
+                                    break;
+                                default:
+                                    w.write("    // Unsupported type_definition for %", fieldName);
+                                    break;
+                                }
+                            },
+                            [&](generic_type_index)
+                            {
+                                w.write("    // TODO: generic_type_index for %", fieldName);
+                            },
+                            [&](generic_type_instance const& td)
+                            {
+                                XLANG_ASSERT(td.generic_args.size() == 1);
+                                call(td.generic_args[0],
+                                    [&](fundamental_type const& gtd)
+                                    {
+                                        w.write("    ABI.System.%Marshaller.UnboxToManaged((void*)value.%)", to_string(gtd), fieldName);
+                                    },
+                                    [&](auto const&) { w.write("    // TODO: Handle generic_type_instance for other non fundamental_type types"); }
+                                );
+                            },
+                            [&](generic_type_param)
+                            {
+                                w.write("    // TODO: generic_type_param for %", fieldName);
+                            },
+                            [&](fundamental_type const& type)
+                            {
+                                if (type == fundamental_type::Boolean)
+                                {
+                                    w.write("    value.% != 0", fieldName);
+                                }
+                                else if (type == fundamental_type::Char)
+                                {
+                                    w.write("    (char)value.%", fieldName);
+                                }
+                                else if (type == fundamental_type::String)
+                                {
+                                    w.write("    HStringMarshaller.ConvertToManaged((void*)value.%)", fieldName);
+                                }
+                                else
+                                {
+                                    w.write("    value.%", fieldName);
+                                }
+                            });
+                    }, ",\n", type.FieldList()));
+        }
+
+        w.write(
```

Comment:
Don't we also need the `Dispose` methods for non blittable struct types that contain managed fields?

@manodasanW we said those would go here in each marshaller class, right? 🤔

Reply:
Ah WindowsRuntimeValueTypeMarshaller.BoxToUnmanaged won't handle it?
We would need to handle it here instead?